src/main/java/org/zanata/mt/api/TranslateResource.java

```diff
@@ -0,0 +1,274 @@
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(response).build();
+        }
+
+        log.debug("Request translations:" + article + " source_lang:"
+            + sourceLang + " target_lang" + targetLang + " provider:"
+            + providerString);
+
+        LocaleId searchLocale = getLocaleFromMap(sourceLang);
+        Locale srcLocale = localeDAO.getOrCreateByLocaleId(searchLocale);
+
+        searchLocale = getLocaleFromMap(targetLang);
+        Locale transLocale = localeDAO.getOrCreateByLocaleId(searchLocale);
+
+        org.zanata.mt.model.Document doc = documentDAO
+            .getOrCreateByUrl(article.getUrl(), srcLocale, transLocale);
+
+        try {
+            String translatedTitle =
+                    translationService.translate(article.getTitle(), srcLocale,
+                            transLocale, provider);
+
+            String translatedBody =
+                    translateBody(article.getDivContent(), srcLocale,
+                            transLocale, provider);
+
+            Article newArticle =
+                new Article(translatedTitle, translatedBody, article.getUrl());
+
+            doc.incrementCount();
+            documentDAO.persist(doc);
+            return Response.ok().entity(newArticle).build();
+        } catch (BadTranslationRequestException e) {
+            String title = "Bad request";
+            log.error(title , e);
+            APIErrorResponse response =
+                new APIErrorResponse(Response.Status.BAD_REQUEST, e, title);
+            return Response.status(Response.Status.BAD_REQUEST).entity(response)
+                .build();
+        } catch (TranslationEngineException e) {
+            String title = "Unable to get translations from engine";
+            log.error(title, e);
+            APIErrorResponse response =
+                new APIErrorResponse(Response.Status.BAD_REQUEST, e, title);
+            return Response.serverError().entity(response).build();
+        }
+    }
+
+    private String translateBody(String body, Locale srcLocale,
+            Locale transLocale, Provider provider)
+            throws TranslationEngineException, BadTranslationRequestException {
+        Document document = Jsoup.parse(body);
+        document = translateArticleHeader(document, srcLocale, transLocale,
+                provider);
+        document = translateArticleBody(document, srcLocale, transLocale,
+                provider);
+        return TranslationUtil.getBodyHTMLContent(document);
+    }
+
+    private Document translateArticleHeader(Document document, Locale srcLocale,
+            Locale transLocale, Provider provider)
+            throws TranslationEngineException, BadTranslationRequestException {
+        Elements header = document.getElementsByTag("header");
+        Element content = header.select("h1.title").first();
+        Element secondaryContent = header.select("span.status").first();
```

Comment:
Might be good to keep this content specific logic out of the REST class. Perhaps use some functional programming here to you can easily change how to extract the content.